drivers/serial/uart_nrfx_uarte.c

```diff
@@ -571,6 +621,10 @@ static void tx_start(const struct device *dev, const uint8_t *buf, size_t len)
 	}
 #endif
 
+	if (IS_ENABLED(UARTE_ANY_CACHE) && (config->flags & UARTE_CFG_FLAG_CACHEABLE)) {
+		sys_cache_data_flush_range((void *)buf, len);
```

Comment:
theoretically you could indiscriminately use `dmm_buffer_out_prepare` that would flush the cache for you and be transparent for the legacy targets. But I guess this could be purposefully omitted for performance reasons